### PR TITLE
Implement `addAbortListener` and `getMaxListeners` in `node:events`

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Implement `addAbortListener` and `getMaxListeners` in `node:events`

Add a couple tests. 

@paperdave why didn't you add this initially? It seems pretty small

### How did you verify your code works?

A couple tests

### CODE REVIEW

File: docs/runtime/nodejs-apis.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Some methods are not optimized yet.
 
 ### [`node:events`](https://nodejs.org/api/events.html)
 
-🟡 Missing `addAbortListener` `events.getMaxListeners`
+🟢 Fully implemented.
```
Comment (paperclover) [inline docs/runtime/nodejs-apis.md]: `events.getMaxListeners` does not support passing `EventTarget`

**Review by paperclover — CHANGES_REQUESTED** [review]
documentation nitpick.

File: docs/runtime/nodejs-apis.md (hunk 1 above)
Comment (Jarred-Sumner) [inline docs/runtime/nodejs-apis.md]:
```suggestion
🟡 `events.addAbortListener` `events.getMaxListeners` do not support (web api) `EventTarget`
```

File: docs/runtime/nodejs-apis.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ Some methods are not optimized yet.
 
 ### [`node:events`](https://nodejs.org/api/events.html)
 
-🟡 Missing `addAbortListener` `events.getMaxListeners`
+🟡 `events.addAbortListener` `events.getMaxListeners` do not support (web api) `EventTarget`
```
Comment (Jarred-Sumner) [inline docs/runtime/nodejs-apis.md]:
```suggestion
🟡 `events.addAbortListener` & `events.getMaxListeners` do not support (web api) `EventTarget`
```

**Pull request merged**